src/networking.c

```diff
@@ -419,8 +396,20 @@ void _addReplyToBufferOrList(client *c, const char *s, size_t len) {
         return;
     }
 
-    size_t reply_len = _addReplyToBuffer(c,s,len);
-    if (len > reply_len) _addReplyProtoToList(c,c->reply,s+reply_len,len-reply_len);
+    /* We update the buffer peak always */
+    const size_t available = c->buf_usable_size - c->bufpos;
+    c->buf_peak = max(c->buf_peak,(size_t)available);
```

Comment:
seems this line is not correct, we shouldn't udpate peak when we haven't written any data to c->buf.
failed CI: https://github.com/sundb/redis/actions/runs/12085859438/job/33703969472